src/Operation/ModifyCollection.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/*
+ * Copyright 2018 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Server;
+use MongoDB\Driver\Session;
+use MongoDB\Driver\WriteConcern;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Exception\InvalidArgumentException;
+
+/**
+ * Operation for the collMod command.
+ *
+ * @api
+ * @see \MongoDB\Database::modifyCollection()
+ * @see http://docs.mongodb.org/manual/reference/command/collMod/
+ */
+class ModifyCollection implements Executable
+{
+    private $databaseName;
+    private $collectionName;
+    private $collModOption;
+    private $collModValue;
+    private $options;
+
+    /**
+     * Constructs a collMod command.
+     *
+     * Supported options:
+     *
+     *  * session (MongoDB\Driver\Session): Client session.
+     *
+     *    Sessions are not supported for server versions < 3.6.
+     *
+     *  * typeMap (array): Type map for BSON deserialization. This will only be
+     *    used for the returned command result document.
+     *
+     *  * writeConcern (MongoDB\Driver\WriteConcern): Write concern.
+     *
+     *    This is not supported for server versions < 3.2 and will result in an
+     *    exception at execution time if used.
+     *
+     * @param string       $databaseName       Database name
+     * @param string       $collectionOrView   Collection or view to modify
+     * @param string       $collModOptionValue Options and values for collMod
+     * @param array        $options            Command options
+     * @throws InvalidArgumentException for parameter/option parsing errors
+     */
+    public function __construct($databaseName, $collectionOrView, $collModOptionValue, array $options = [])
+    {
```

Comment:
Before we validate options, I think we can add a small check to ensure that the `$collectionOptions` array is not empty, as is done for `$operations` in BulkWrite.

In my tests, MongoDB 3.6 lets us run the command `{ "collMod": "foo" }` with no additional options successfully, but I have to assume that's a no-op. You can double-check with @ksuarz and/or @visemet, but think it'd make sense to yell at our users for trying to do that. If anything, it'd help them realize they're inadvertently running a no-op.

Reply:
I looked at the {{collMod}} code in the server; it does what you'd expect in that the options for the collection or view are not changed. But it doesn't look like we optimize it to be a "no-op"; my reading of the code is that it will still commit a metadata write to the storage layer.

I don't think it really matters either way; a {{collMod}} with no options is still a valid command and the server will just do what you tell it to do.

Reply:
Understandable that it's not technically a no-op on the server if it commits. Thanks for looking into it.

@kvwalker: Let's still throw here.